### PR TITLE
OPRUN-3402: :warning: Bump OLM API to v1

### PR DESCRIPTION
**Pull request by perdasilva**
* Promotes operator/v1alpha1 OLM to operator/v1 OLM

Next PR: #2066

### CODE REVIEW

**Comment by perdasilva** [comment]
verify-crd-schema is failing because `observedGeneration` is not in `OperatorCondition`. We've reached out on the #forum-api-review to see if this is something we should override. [slack thread](https://redhat-internal.slack.com/archives/CE4L0F143/p1729020888154599)

**Comment by JoelSpeed** [comment]
@perdasilva You need to ensure your project is checked out under an appropriate `$GOPATH` or, not run the protobuf generator 

Looking at the PR, I think protobuf generation is not required, so perhaps revert all the proto related changes and run the updates with `PROTO_OPTIONAL=1`

**Comment by perdasilva** [comment]
Thanks @JoelSpeed ^^ I just saw the message on the api forum XD - should be fixed up now

**Comment by perdasilva** [comment]
@JoelSpeed This should be ready for review ^^

The failure in verify-crd-schema is:

```
 	could not run schemacheck generator for group/version operator.openshift.io/v1: 
		error in operator/v1/zz_generated.crd-manifests/0000_10_operator-lifecycle-manager_01_olms.crd.yaml: ConditionsMustHaveProperSSATags: crd/olms.operator.openshift.io version/v1 field/^.status.condition must define valid condition properties: observedGeneration attribute is missing
		error in operator/v1/zz_generated.featuregated-crd-manifests/olms.operator.openshift.io/AAA_ungated.yaml: ConditionsMustHaveProperSSATags: crd/olms.operator.openshift.io version/v1 field/^.status.condition must define valid condition properties: observedGeneration attribute is missing
+ echo This verifier checks all files that have changed. In some cases you may have changed or renamed a file that already contained api violations, but you are not introducing a new violation. In such cases it is appropriate to /override the failing CI job.
This verifier checks all files that have changed. In some cases you may have changed or renamed a file that already contained api violations, but you are not introducing a new violation. In such cases it is appropriate to /override the failing CI job. 
```

as discussed [here](https://redhat-internal.slack.com/archives/CE4L0F143/p1729020888154599), it should be ok to override.

(cc @joelanford)

File: openshift
[the remark this answers is not in the document]
Comment (perdasilva) [reply openshift:1]: :man_facepalming:

[the remark this answers is not in the document]
Comment (perdasilva) [reply openshift:1]: Okie - all fixed! Thanks, and sorry I missed that =SS

**Comment by JoelSpeed** [comment]
When bumping an API from v1alpha1 to v1, we must keep the old types around for some time. We need to be able to generate clients for both v1alpha1 and v1 simultaneously to allow users to build tooling versions that might span both GA and pre-GA

Please restore the v1alpha1 versions, and then I'll review the v1 API again to see if there's anything we might need to update prior to GA

**Comment by perdasilva** [comment]
> When bumping an API from v1alpha1 to v1, we must keep the old types around for some time. We need to be able to generate clients for both v1alpha1 and v1 simultaneously to allow users to build tooling versions that might span both GA and pre-GA
> 
> Please restore the v1alpha1 versions, and then I'll review the v1 API again to see if there's anything we might need to update prior to GA

@JoelSpeed Thanks for clarifying that. We really weren't sure if it would be ok given the support requirements, but didn't factor in other technical considerations. I've gone ahead and restored v1alpha1.

File: operator/v1/types_olm.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+// OLM provides information to configure an operator to manage the OLM controllers
+//
+// Compatibility level 1: Stable within a major release for a minimum of 12 months or 3 minor releases (whichever is longer).
+// +openshift:compatibility-gen:level=1
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=olms,scope=Cluster
+// +kubebuilder:subresource:status
+// +openshift:api-approved.openshift.io=https://github.com/openshift/api/pull/1504
+// +openshift:file-pattern=cvoRunLevel=0000_10,operatorName=operator-lifecycle-manager,operatorOrdering=01
+// +kubebuilder:validation:XValidation:rule="self.metadata.name == 'cluster'",message="olm is a singleton, .metadata.name must be 'cluster'"
```
Comment (JoelSpeed) [inline operator/v1/types_olm.go:18]: Has this been tested and we are happy that it works?

**Comment by JoelSpeed** [comment]
Is there a related feature gate that also needs to be promoted? 

What is the path to GA for this feature?

Do we have enough time to complete the path to GA prior to branch cutting?

File: operator/v1/types_olm.go (hunk 1 above)
Comment (perdasilva) [reply operator/v1/types_olm.go:18]: yup yup - we have a test for it https://github.com/openshift/api/pull/2061/files#diff-34775f0b3bbcc2a64f685c6fc398e13b3fae20851398b2b5a455726630673a01R21

I've also gone ahead and manually tested on a clusterbot. Looks good ^^

**Comment by JoelSpeed** [comment]
/lgtm

/override ci/prow/verify-crd-schema 

Existing failures about observed generation cannot be resolved now

**Pull request merged**